File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxpro = 0;
        int curs = prices[0];
        for (auto& p : prices)
        {
            curs = min(curs, p);
            maxpro = max(maxpro, p - curs);
        }
        return maxpro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxpro = 0
        curs = prices[0]
        for p in prices:
            curs = min(curs, p)
            maxpro = max(maxpro, p - curs)
        return maxpro